dockerclient/client.go

```diff
@@ -694,6 +694,10 @@ func (e *ClientExecutor) UnrecognizedInstruction(step *imagebuilder.Step) error
 // the user command into a shell and perform those operations before. Since RUN
 // requires /bin/sh, we can use both 'cd' and 'export'.
 func (e *ClientExecutor) Run(run imagebuilder.Run, config docker.Config) error {
+	if len(run.Mounts) > 0 {
+		return fmt.Errorf("RUN --mount not supported")
```

Comment:
So we are adding support just to throw an error if the user uses it?

Reply:
Took another look, add support! It should use the same code as what I'm writing in buildah so it'll be fairly easy

Reply:
Wait sorry, took another look at line 767 
if err := e.Client.StartExec(exec.ID, docker.StartExecOptions{...}
From my understanding, imagebuilder run uses docker.StartExecOptions, and that doesn't support the --mounts flag. I could add the mounts to `e.TransientMounts` but that means the mount will be visible for other steps in the build, which is what we don't want when we use the --mount flag with a RUN. 

I'm adding support for this because we can do what I said above in buildah, since we can add a mounts field to buildah's builder.RunOptions.

Reply:
Let me know though if there's a way around it or if I'm reading it wrong

Reply:
update: I might have found a way, let me take another look, sorry for the noise

Reply:
> So we are adding support just to throw an error if the user uses it?

I find it preferable to adding the support for parsing the flag, which we need to do in order to do something with the flag in buildah, and then just ignoring it in the dockerclient.  That'd be how we got #187.

Reply:
Sure, I can just remove the error